File: C#/0141. Linked List Cycle.cs
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int x) {
 *         val = x;
 *         next = null;
 *     }
 * }
 */
public class Solution {
    public bool HasCycle(ListNode head) {
        if (head==null || head.next==null){
            return false;
        }
        ListNode pslow=head;
        ListNode pfast=head;
        while(pfast.next!=null && pfast.next.next!=null){
            pslow=pslow.next;
            pfast=pfast.next.next;
            if (pslow==pfast){
                return true;
            }
        }
        return false;
    }
}

File: 20200215Python-China/0141. Linked List Cycle.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def hasCycle(self, head: ListNode) -> bool:
        node=ListNode(0)
        node.next=head
        pFast,pSlow=node,node
        while pFast.next and pFast.next.next:
            pFast=pFast.next.next
            pSlow=pSlow.next
            if pFast==pSlow:
                return True
        return False
